coucharchive

```diff
@@ -307,10 +307,22 @@ class ReplicationControl(object):
         # Choose the value to achieve replication in ideal_duration:
         ideal_number = self._ideal_number_of_replications_for_ideal_duration()
 
-        # ... but if there were errors, take 0.9 × the lowest error value
-        if self._last_errors:
-            min_number_with_error = min(e[1] for e in self._last_errors)
-            ideal_number = int(0.9 * min_number_with_error)
+        # ... but if there were errors, compute a age weighted average
+        # between past erros and ideal_number.
+        now = time.time()
+        weights = []
+        weighted_values = []
+        target = [(now, ideal_number)]
+        for i in self._last_errors + target:
+            age = i[0] - now
+            # Weight for an old event is 0, weight for a recent event is 1,
+            # between the two it is a linear curve.
+            # 0.0033 = 1 / (5 * 60)
+            weight = max(0, 0.0033 * age + 1)
```

Comment:
In my opinion it would be as clear as currently with:
```python
            # Weight for an old event is 0, weight for a recent event is 1,
            # in-between it's a linear curve:
            weight = max(0, 1 + age / (5 * 60))
```

Reply:
PS: Maybe it's me, but at first I did not understand because I expected `age > 0` (it's the opposite actually).

Maybe this would be more understandable?
```python
age = now - i[0]
weight = max(0, 1 - age / (5 * 60))
```

Reply:
Indeed, surely, thanks!